### PR TITLE
Tweak/Experiments Dom Optimization - ED-544

### PR DESCRIPTION
**Pull request by kobizz**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/editor/editor.php [hunk 1]
```diff
@@ -526,6 +526,16 @@ public function enqueue_scripts() {
...
+		$active_experimental_features = [];
+
+		$experiments = Plugin::$instance->experiments;
+
+		foreach ( $experiments->get_features() as $feature_name => $feature ) {
+			if ( $experiments->is_feature_active( $feature_name ) ) {
+				$active_experimental_features[ $feature_name ] = true;
+			}
+		}
+
```
Comment (matipojo) [inline core/editor/editor.php:538]: Move the a method `get_active_features`.

File: core/experiments/manager.php [hunk 2]
```diff
@@ -76,6 +80,17 @@ public function add_feature( array $options ) {
 
 		$this->features[ $options['name'] ] = $experimental_data;
 
+		if ( is_admin() ) {
+			$feature_option_key = $this->get_feature_option_key( $options['name'] );
+
+			$on_state_change_callback = function( $old_state, $new_state ) use ( $experimental_data ) {
```
Comment (matipojo) [inline core/experiments/manager.php:86]: It look like the `feature` should be a class that handles his state change and other aspects.

File: core/experiments/manager.php [hunk 3]
```diff
@@ -164,6 +175,17 @@ private function get_feature_option_key( $feature_name ) {
 		return 'elementor_experiment-' . $feature_name;
 	}
 
+	private function add_default_features() {
+		$this->add_feature( [
+			'name' => 'e-dom_optimization',
```
Comment (matipojo) [inline core/experiments/manager.php:180]:
```suggestion
			'name' => 'e_dom_optimization',
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/editor/editor.php [hunk 4]
```diff
@@ -588,10 +592,7 @@ public function enqueue_scripts() {
 				'darkModeStylesheetURL' => ELEMENTOR_ASSETS_URL . 'css/editor-dark-mode' . $suffix . '.css',
 				'defaultGenericFonts' => $kits_manager->get_current_settings( 'default_generic_fonts' ),
 			],
-			// Legacy Mode - for backwards compatibility of older HTML markup.
-			'legacyMode' => [
-				'elementWrappers' => Plugin::instance()->get_legacy_mode( 'elementWrappers' ),
-			],
+			'experimentalFeatures' => $active_experimental_features,
```
[the remark this answers is not in the document]
Comment (kobizz) [reply core/editor/editor.php:595]: aca3205

**Review by matipojo — DISMISSED** [review]

**Review by KingYes — DISMISSED** [review]

**Pull request merged**